backend/src/main/java/org/eclipse/tractusx/puris/backend/controller/PartnerController.java

```diff
@@ -0,0 +1,203 @@
+        }
+        return new ResponseEntity<>(modelMapper.map(partnerEntity, PartnerDto.class), HttpStatusCode.valueOf(200));
+    }
+
+    @PutMapping("putAddress")
+    @CrossOrigin
+    @Operation(description = "Updates an existing Partner by adding a new Address. If that this Partner already has " +
+        "an Address with the BPNA given in the request body, that existing Address will be overwritten. " +
+        "You have to provide either the BPNL parameter or the UUID parameter in order to identify the requested Partner." +
+        " If both are provided, the BPNL is evaluated first.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Update accepted"),
+        @ApiResponse(responseCode = "400", description = "Invalid Address data"),
+        @ApiResponse(responseCode = "404", description = "Partner not found"),
+        @ApiResponse(responseCode = "500", description = "")
+    })
+    public ResponseEntity<?> addAddress(@RequestParam(required = false) String bpnl,
+                                        @RequestParam(required = false) UUID uuid,
+                                        @RequestBody AddressDto address) {
+        Partner existingPartner = findPartner(bpnl, uuid);
+        if (existingPartner == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(404));
+        }
+        Address newAddress;
+        try {
+            newAddress = modelMapper.map(address, Address.class);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(500));
+        }
+        // Remove operation in case there is an update of an existing address
+        existingPartner.getAddresses().remove(newAddress);
+
+        existingPartner.getAddresses().add(newAddress);
+        existingPartner = partnerService.update(existingPartner);
+        if (existingPartner == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        return new ResponseEntity<>(HttpStatusCode.valueOf(200));
+    }
+
+    @PutMapping("putSite")
+    @CrossOrigin
+    @Operation(description = "Updates an existing Partner by adding a new Site. If that this Partner already has " +
+        "a Site with the BPNS given in the request body, that existing Site will be overwritten. " +
+        "You have to provide either the BPNL parameter or the UUID parameter in order to identify the requested Partner." +
+        " If both are provided, the BPNL is evaluated first.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Update accepted"),
+        @ApiResponse(responseCode = "400", description = "Invalid Address data"),
+        @ApiResponse(responseCode = "404", description = "Partner not found"),
+        @ApiResponse(responseCode = "500", description = "")
+    })
+    public ResponseEntity<?> addSite(@RequestParam(required = false) String bpnl,
```

Comment:
same as above. But great idea to either use uuid or bpnl

Reply:
Annotations and examples added.